### DISCUSSION TITLE
Next.js 13.4.2: Navigation between pages in `/app` and `/pages` folders causes full app reload

### DISCUSSION THREAD

**Discussion by anhdd-kuro** [vercel/next.js#49849] [App Router]
I am incrementally migrating from `pages` to `app` and currently facing this problem
I have

```
/app ( both server component by default )
  /foo
  /bar

/pages
  /foo-page.tsx
  /bar-page.tsx
```

With simple client component navigation like this

```ts
"use client";
import Link from "next/link";

export const CNavBar: React.FC = () => {
  return (
    <nav>
        <Link href="/foo">foo</Link>
        <Link href="/bar">bar</Link>
        <Link href="/foo-page">foo-page</Link>
        <Link href="/bar-page">bar-page</Link>
    </nav>
  );
};

```

There is no problem when navigate between `foo` and `bar` which is in `app` and also no problem when navigate between `foo-page` and `bar-page` which is in `pages`
But when navigate between `foo` and `foo-page` and so on , It's causing full reload
Am I missing something ?
Look like I'm not the only one : [stackoverflow](https://stackoverflow.com/questions/76207187/next-js-13-4-2-navigation-between-pages-in-app-and-pages-folders-causes-f)

**Comment by icyJoseph** [comment]
AFAIK, this is how it is expected to be... but I can't really build a good rationale, other than, the rendering of app and pages is different, meaning it creates a different React tree, that handles serialized data as React Server Components data, and there's different router engines on both. 

Like navigating between both, is not just removing a couple of divs from the DOM, it is a whole lot more than that. There's client side caching layers, routing is different, script tags needed are different, the window globals are different too (I think) ~

It'd be nice to have some documentation stating that hard reload will happen, I couldn't find it.

**Reply by pouyarezvani** [reply]
Are there any updates on this
